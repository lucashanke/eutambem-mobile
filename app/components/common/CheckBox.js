import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';

import ExternalCheckBox from 'react-native-check-box';
import { DARK_PURPLE } from '../../styles';
import input from '../hoc/input';

const styles = StyleSheet.create({
  checkbox: {
    paddingTop: 10,
  },
});

const CheckBox = props => (
  <ExternalCheckBox
    {...props}
    onClick={props.onValueChange}
    isChecked={props.value}
    style={styles.checkbox}
    checkBoxColor={DARK_PURPLE}
    rightText={props.label}
  />
);

CheckBox.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.bool,
};

CheckBox.defaultProps = {
  value: false,
};

export default input(CheckBox);
