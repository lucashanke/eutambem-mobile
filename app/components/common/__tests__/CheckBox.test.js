import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import { CheckBox } from '../CheckBox';
import ExternalCheckBox from 'react-native-check-box';

const props = {
  label: 'Test CheckBox',
  onValueChange: sinon.spy(),
  value: true,
};

describe('CheckBox', () => {
  it('renders external checkbox component', () => {
    const wrapper = shallow(<CheckBox {...props} />);
    expect(wrapper.find(ExternalCheckBox)).toHaveLength(1);
  });

  it('renders title as rightText on external checkbox component', () => {
    const wrapper = shallow(<CheckBox {...props} />);
    const checkbox = wrapper.find(ExternalCheckBox);
    expect(checkbox.props().rightText).toEqual('Test CheckBox');
  });

  it('passes down value to the external checkbox component as isChecked', () => {
    const wrapper = shallow(<CheckBox {...props} />);
    const checkbox = wrapper.find(ExternalCheckBox);
    expect(checkbox.props().isChecked).toEqual(true);
  });

  it('calls onValueChange when external checkbox component is clicked', () => {
    const wrapper = shallow(<CheckBox {...props} />);
    wrapper.find(ExternalCheckBox).simulate('click');
    expect(props.onValueChange.callCount).toEqual(1);
  });
});
