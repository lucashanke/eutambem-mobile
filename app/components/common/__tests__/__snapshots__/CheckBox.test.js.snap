// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox renders correctly 1`] = `
<ForwardRef(View)
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <ForwardRef(View)
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <CheckBox
      label="Test CheckBox"
      onClick={[Function]}
      onValueChange={[Function]}
      required={false}
      value={false}
    />
  </ForwardRef(View)>
</ForwardRef(View)>
`;
